src/mrpro/operators/WaveletOp.py

```diff
@@ -0,0 +1,374 @@
+"""Wavelet operator."""
+
+from collections.abc import Sequence
+from typing import Literal
+
+import numpy as np
+import torch
+from ptwt.conv_transform import wavedec, waverec
+from ptwt.conv_transform_2 import wavedec2, waverec2
+from ptwt.conv_transform_3 import wavedec3, waverec3
+from pywt import Wavelet
+from pywt._multilevel import _check_level
+
+from mrpro.operators.LinearOperator import LinearOperator
+
+# Switch off formatter to avoid having only one wavelet type per line
+# fmt: off
+WaveletType = Literal[
```

Comment:
Consider using a dictionary or a more structured data type for managing wavelet types rather than a long list of literals. This could improve readability and maintainability. [important]